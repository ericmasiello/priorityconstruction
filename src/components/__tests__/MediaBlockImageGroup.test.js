import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import StyledMediaBlockImageGroup, { MediaBlockImageGroup } from '../MediaBlockImageGroup';

it('should render', () => {
  const wrapper = shallow(<MediaBlockImageGroup />);

  expect(wrapper).toHaveLength(1);
});

it('should render default styles', () => {
  const component = renderer.create(<StyledMediaBlockImageGroup />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('should set grid styles when rendering more than one image', () => {
  const component = renderer.create(<StyledMediaBlockImageGroup size={2} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render an apporpriate number of grid rows', () => {
  const component = renderer.create(<StyledMediaBlockImageGroup size={10} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render styles for the last image when displaying an odd number of images', () => {
  const component = renderer.create(<StyledMediaBlockImageGroup size={3} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
