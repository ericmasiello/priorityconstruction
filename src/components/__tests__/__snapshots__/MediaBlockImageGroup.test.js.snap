// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an apporpriate number of grid rows 1`] = `
.c0 {
  display: grid;
  grid-gap: 0.5rem;
  position: relative;
  grid-template-rows: repeat(5,1fr);
  grid-template-columns: repeat(2,1fr);
}

.c0 .c1 {
  height: 100%;
}

<div
  className="c0"
/>
`;

exports[`should render default styles 1`] = `
.c0 .c1 {
  height: 100%;
}

<div
  className="c0"
/>
`;

exports[`should render styles for the last image when displaying an odd number of images 1`] = `
.c0 {
  display: grid;
  grid-gap: 0.5rem;
  position: relative;
  grid-template-rows: repeat(2,1fr);
  grid-template-columns: repeat(2,1fr);
}

.c0 .c1 {
  height: 100%;
}

.c0 > :last-child {
  grid-row: 2;
  position: absolute !important;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100%;
}

<div
  className="c0"
/>
`;

exports[`should set grid styles when rendering more than one image 1`] = `
.c0 {
  display: grid;
  grid-gap: 0.5rem;
  position: relative;
  grid-template-rows: repeat(1,1fr);
  grid-template-columns: repeat(2,1fr);
}

.c0 .c1 {
  height: 100%;
}

<div
  className="c0"
/>
`;
